src/common/types/types.cpp

```diff
@@ -707,12 +707,19 @@ bool LogicalTypeUtils::isNumerical(const LogicalType& dataType) {
 }
 
 bool LogicalTypeUtils::isNested(const LogicalType& dataType) {
-    switch (dataType.typeID) {
+    return isNested(dataType.typeID);
+}
+
+bool LogicalTypeUtils::isNested(kuzu::common::LogicalTypeID logicalTypeID) {
```

Comment:
Guess, we can tell whether a datatype is nested by its physical type.

Reply:
Technically possible but our function is registered with LogicalTypeID rather than LogicalType at this point. So it's hard to make the change now.